### ISSUE TITLE
knife vsphere vm find

### ISSUE THREAD

**Issue by coda4096** [chef-partners/knife-vsphere#349]
### What are you trying to do and what's not working?
I am trying to get the host (hypervisor) that a VM is running on in a cluster

### Please paste the command and output (run with `-VV`). Don't forget to X out any passwords or sensitive information
```
knife vsphere vm find --pool MY_POOL_NAME --match-name MY_VM_NAME
```

### Other details, such as gem version and if you're working with a Linux or Windows VM
When a VM is created and started in a cluster, Distributed Resource Scheduler (DRS) will put the vm wherever it decides. I would like to be able to see where that vm was started. (does this make sense?)

**Comment by swalberg** [comment]
Looks like we can get to it from [VirtualMachine](https://www.vmware.com/support/developer/vc-sdk/visdk41pubs/ApiReference/vim.VirtualMachine.html) -> [summary](https://www.vmware.com/support/developer/vc-sdk/visdk41pubs/ApiReference/vim.vm.Summary.html) -> [runtime](https://www.vmware.com/support/developer/vc-sdk/visdk41pubs/ApiReference/vim.vm.RuntimeInfo.html) -> host

What are you trying to use this for? At least in our environment, DRS will vMotion guests around even after cloning to make sure there are no hot spots. Just trying to make sure we're getting you the actual information you need.

**Comment by coda4096** [comment]
We mainly want to use it for remote access, ie. vnc uses the external IP for the esx host that the vm lives in.

**Comment by coda4096** [comment]
**vm.summary.runtime.host** seems to give output like this 

> HostSystem("host-15")	

it would be more useful if we can get the host IP or hostname

**Comment by swalberg** [comment]
Right.. that gets you to [HostSystem](https://www.vmware.com/support/developer/vc-sdk/visdk41pubs/ApiReference/vim.HostSystem.html). Try `summary.host`? [summary](https://www.vmware.com/support/developer/vc-sdk/visdk41pubs/ApiReference/vim.host.Summary.html) gets you some info on the host itself.

**Comment by swalberg** [comment]
or try `vm.summary.runtime.host.name`. It's not obvious from the page I linked but it turns out HostSystem inherits from ManagedEntity which has a `name` property.

**Comment by coda4096** [comment]
Yes! that gives me an IP, this is useful

**Comment by swalberg** [comment]
Which one? Also, I noticed this command doesn't use the output format we've been moving to, which would let you script this nicely with JSON output `knife vsphere vm find --pool MY_POOL_NAME --match-name MY_VM_NAME --format json`. I'll get that on my list.

**Comment by coda4096** [comment]
vmc.summary.runtime.host.name gives me the host's IP

**Comment by coda4096** [comment]
I will put together a pull request.  Is the field **--host-ip** going to be a good one to use

**Comment by swalberg** [comment]
Could you use `--host-name` instead, please? According to the API it's a name (mine shows a FQDN, not an IP) I suspect it has to do with how you registered the host initially.

**Comment by coda4096** [comment]
no worries, did you mean --hostname that exists or create --host-name instead?

**Comment by coda4096** [comment]
pull request done.
